developer/src/kmcmpdll/Compiler.cpp

```diff
@@ -330,66 +351,74 @@ extern "C" BOOL __declspec(dllexport) CompileKeyboardFile(PSTR pszInfile, PSTR p
   currentLine = 0;
   nErrors = 0;
 
-  hInfile = CreateFileA(pszInfile, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, 0, NULL);
-  if (hInfile == INVALID_HANDLE_VALUE) SetError(CERR_InfileNotExist);
+  fp_in = fopen((const char*)pszInfile, "rb");
+
+  if (fp_in == NULL) SetError(CERR_InfileNotExist);
+
 
   // Transfer the file to a memory stream for processing UTF-8 or ANSI to UTF-16?
   // What about really large files?  Transfer to a temp file...
-
-  if (!ReadFile(hInfile, str, 3, &len, NULL))
+  fseek(fp_in, 0, SEEK_END);
+  len = ftell(fp_in);
+  fseek(fp_in, 0, SEEK_SET);
+  if (!fread(str, 1, 3, fp_in))
   {
-    CloseHandle(hInfile);
+    fclose(fp_in);
     return CERR_CannotReadInfile;
   }
-  SetFilePointer(hInfile, 0, NULL, FILE_BEGIN);
+
+  fseek(fp_in, 0, SEEK_SET);
   if (str[0] == UTF8Sig[0] && str[1] == UTF8Sig[1] && str[2] == UTF8Sig[2])
-    hInfile = UTF16TempFromUTF8(hInfile, TRUE);
+    fp_in = UTF16TempFromUTF8(fp_in, TRUE);
   else if (str[0] == UTF16Sig[0] && str[1] == UTF16Sig[1])
-    SetFilePointer(hInfile, 2, NULL, FILE_BEGIN);
+    fseek(fp_in, 2, SEEK_SET);
   else
-    hInfile = UTF16TempFromUTF8(hInfile, FALSE);  // Will fall back to ansi for invalid UTF-8
-  if (hInfile == INVALID_HANDLE_VALUE)   // I3228   // I3510
+    fp_in = UTF16TempFromUTF8(fp_in, FALSE);
+  if (fp_in == NULL)
   {
     return CERR_CannotCreateTempfile;
   }
 
-  hOutfile = CreateFileA(pszOutfile, GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, 0, NULL);
-  if (hOutfile == INVALID_HANDLE_VALUE) SetError(CERR_CannotCreateOutfile);
+  fp_out = fopen((const char*)pszOutfile, "wb");
+
+  if (fp_out == NULL) SetError(CERR_CannotCreateOutfile);
+
 
-  DWORD msg;
+  KMX_DWORD msg;
   FILE_KEYBOARD fk;
   CodeConstants = new NamedCodeConstants;
-
-  err = CompileKeyboardHandle(hInfile, &fk);
+  err = CompileKeyboardHandle(fp_in, &fk);
   if (err)
   {
-    if ((msg = WriteCompiledKeyboard(&fk, hOutfile)) != CERR_None)
+    if ((msg = WriteCompiledKeyboard(&fk, fp_out)) != CERR_None)
```

Comment:
msg is called like that everywhere (engine, win,...) - for consistency I`like to keep that name.  It`s also much more readable in 1 line since we use msg in a lot of places in the same way